openfl/utilities/attestation/attestation_utils.py

```diff
@@ -0,0 +1,259 @@
+            attestation_report_path (str): Path to the store attestation reports.
+            ita_api_key (str): API key for ITA.
+            avs_url (str): URL for AVS.
+            enable_remote_attestation (bool): Flag to enable remote attestation.
+        """
+        self.participant_name = participant_name
+        self.attestation_report_path = attestation_report_path
+        self.ita_api_key = ita_api_key
+        self.avs_url = avs_url
+
+
+    def get_attestation_report(
+            self
+    ):
+        """Generates the attestation report for the enclave.
+        This function generates the SGX quote, fetches the MRENCLAVE value,
+        and obtains the attestation report from AVS.
+        It also generates an ECDSA-P 384 key pair and certificate for mTLS
+        with the collaborator enclave.
+        The generated report is saved in the specified attestation report path.
+        The function also checks if the required environment variables are set
+        and raises an exception if any of them are missing.
+        The function returns a dictionary containing the MRENCLAVE value,
+        participant name, and the AVS report token.
+        The function raises a FileNotFoundError if the attestation report path
+        does not exist, and a ValueError if the ITA API key or AVS URL is not set.
+        Args:
+            None
+        Raises:
+            FileNotFoundError: If the attestation report path does not exist.
+            ValueError: If the ITA API key or AVS URL is not set.
+        Returns:
+            dict: A dictionary containing the MRENCLAVE value, participant name,
+                  and AVS report token.
+        """
+
+        if not os.path.exists(self.attestation_report_path):
+            raise FileNotFoundError(f"attestation report path {self.attestation_report_path} does not exist")
+        
+        if self.ita_api_key is None:
+            raise ValueError("ITA API key is required for remote attestation") 
+        if self.avs_url is None:
+            raise ValueError("AVS URL is required for remote attestation")
+        
+        mrenclave = self.fetch_mrenclave_from_quote(None)
+        logger.info(f"Enclave MRENCLAVE: {mrenclave}")
+
+        # Generate ECDSA-P 384 key pair and certificate
+        # This is the enclave specific ECDSA-P 384 keypair for setting up mTLS
+        # with collaborator enclave
+        priv_key_path = os.path.join(self.attestation_report_path, f"{self.participant_name}_privkey.pem")
+        ecdsa_p_384_signer = ECDSASigner.get_instance()
+        cert = ecdsa_p_384_signer.cert("localhost",mrenclave).encode('utf-8')
+        key = ecdsa_p_384_signer.serialize_private_key()
+        pub_key = ecdsa_p_384_signer.get_pubkey()
+        pubkey_path = os.path.join(self.attestation_report_path, f"{self.participant_name}_pubkey.pem")
+        with open(pubkey_path, "wb") as fh:
+            fh.write(pub_key.encode('utf-8'))
```

Comment:
I suggest keeping doing this outside of the method, based on the `AttestedIdentity` object.

Reply:
removed. Added a method in AttestedIdentity to save.